### DISCUSSION TITLE
Show off your music

### DISCUSSION THREAD

**Discussion by orhun** [orhun/linuxwave#1] [Show and tell]
Share the masterpieces you have created using linuxwave!

**Comment by Arcmyx** [comment]
Hi! I just wanted to say, as someone with perfect pitch and someone who is into music production, I honestly think this tool is really cool. The fact that it is so modular is great as well, and I can be sure I'm going to use this to inspire melodies in the future.

**Reply by orhun** [reply]
Great to hear that!

**Comment by Arcmyx** [comment]
I am currently working on a GUI for linuxwave! I've implemented tempo shifting, and right now it already auto-adjusts the sample rate and pitch so you get a 44100Hz, C minor sample every time. I will add some more stuff like pitch and an interactive piano to select scale. I just thought you might want to know about it. GitHub repo is at https://github.com/Arcmyx/linuxwavegui.

**Reply by orhun** [reply]
That is awesome! I gave it a spin and it seems like it got stuck at "Generating" for me. (I have `sox` installed). Any idea why?

Also would you be interested in submitting a PR to mention this project in the README.md?

**Reply by Arcmyx** [reply]
Hmmm... It must be failing at window.api.launchApp(). What OS are you on? I am using WSL, so maybe node's child process exec is failing (the function that spawns command shells and executes stuff).

And thank you for letting me add to the README on a new branch! I will attempt to do that now.

**Reply by orhun** [reply]
I'm on Arch Linux.. I don't see anything suspicious in the logs. But the console shows:

> Uncaught TypeError: can't access property "launchApp", window.api is undefined
    sendCustomCommand http://localhost:8000/:35
    onclick http://localhost:8000/:1
[localhost:8000:35:11](http://localhost:8000/)

**Reply by Arcmyx** [reply]
Ah, the issue is that you're using the wrong program. Looks like you're using a browser and an HTTP server. It's not supposed to use a browser; it's supposed to use Node.js and Electron. cd to the directory, make sure 'node' is installed, and then run npm i and npm run start. Sorry, I should have included this in the README.

**Reply by Arcmyx** [reply]
Also npm i should take a while, don't worry if it hangs for ~5min.

**Reply by Arcmyx** [reply]
Oh by the way I also make music for fun (https://www.youtube.com/@ArcmyxOfficial) so I could make a theme song for this project :D

**Reply by orhun** [reply]
Theme song would be absolutely great! Feel free to cook something :D

I just tried running this with NPM and it works :D

<sup>

Btw we have a Discord server with people who are into programming & music, you might be interested: https://grindhouse.dev

</sup>

**Comment by Arcmyx** [comment]
Also can I make it breakcore/dubstep (the theme)?

**Reply by orhun** [reply]
Haha yeah :D
